Sources/TuistCore/Cache/CacheOutputType.swift

```diff
@@ -8,17 +14,65 @@ public enum CacheOutputType: CustomStringConvertible {
     /// Frameworks built for the simulator.
     case framework
 
-    /// XCFrameworks built for the simulator and device.
-    case xcframework
+    /// XCFrameworks built for the simulator and/or device.
+    case xcframework(CacheXCFrameworkDestination?)
 
     public var description: String {
         switch self {
         case .bundle:
             return "bundle"
         case .framework:
             return "framework"
+        case let .xcframework(type):
+            switch type {
+            case .device:
+                return "device-xcframework"
+            case .simulator:
+                return "simulator-xcframework"
+            case nil:
+                return "xcframework"
+            }
+        }
+    }
+}
+
+extension CacheOutputType: Equatable {
```

Comment:
if this logic is only needed by th eCacheXCFrameworkBuilder, define the extension there and as fileprivate